golang/trace/zipkin_tracereporter.go

```diff
@@ -116,20 +116,71 @@ func buildZipkinSpan(span tc.Span, annotations []tc.Annotation, binaryAnnotation
 		ServiceName: targetEndpoint.ServiceName,
 	}
 
-	// TODO Add BinaryAnnotations
 	thriftSpan := tcollector.Span{
-		TraceId:     uint64ToBytes(span.TraceID()),
-		Host:        &host,
-		Name:        targetEndpoint.Operation,
-		Id:          uint64ToBytes(span.SpanID()),
-		ParentId:    uint64ToBytes(span.ParentID()),
-		Annotations: buildZipkinAnnotations(annotations),
-		Debug:       false,
+		TraceId:           uint64ToBytes(span.TraceID()),
+		Host:              &host,
+		Name:              targetEndpoint.Operation,
+		Id:                uint64ToBytes(span.SpanID()),
+		ParentId:          uint64ToBytes(span.ParentID()),
+		Annotations:       buildZipkinAnnotations(annotations),
+		BinaryAnnotations: buildBinaryAnnotations(binaryAnnotations),
+		Debug:             false,
 	}
 
 	return &thriftSpan
 }
 
+func buildBinaryAnnotation(ann tc.BinaryAnnotation) *tcollector.BinaryAnnotation {
+	bann := &tcollector.BinaryAnnotation{Key: ann.Key}
+	switch v := ann.Value.(type) {
+	default:
```

Comment:
let's put the default at the very bottom, and log something if we get to that point.